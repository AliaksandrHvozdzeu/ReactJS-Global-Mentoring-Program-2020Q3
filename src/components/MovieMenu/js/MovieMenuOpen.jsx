import React from 'react';
import PropTypes from "prop-types";

const MovieMenuOpen = ({ action, editMovie, deleteMovie }) => (
  <div className="movie-menu movie-menu-opened">
    <button className="movie-menu-action" type="button" onClick={editMovie}>Edit</button>
    <button className="movie-menu-action" type="button" onClick={deleteMovie}>Delete</button>
    <button className="movie-menu-close" id="movie-menu-close-button" type="button"
            onClick={() => action(false)}>Close
    </button>
  </div>
);

MovieMenuOpen.propTypes = {
  action: PropTypes.func.isRequired,
  editMovie: PropTypes.func.isRequired,
  deleteMovie: PropTypes.func.isRequired
};

export default MovieMenuOpen;
