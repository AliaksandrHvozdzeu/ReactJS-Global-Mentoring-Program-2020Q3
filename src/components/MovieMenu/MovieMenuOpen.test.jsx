import React from "react";
import rerender from "react-test-renderer";
import MovieMenuOpen from "./js/MovieMenuOpen";
import { shallow } from "enzyme";

describe("when MovieMenuOpen", () => {

  const action = () => {
    return false;
  };
  const editMovie = () => {
  };
  const deleteMovie = () => {
  };

  test("then snapshot created", () => {

    const component = rerender.create(<MovieMenuOpen action={action}
                                                     editMovie={editMovie}
                                                     deleteMovie={deleteMovie} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("click open button", () => {
    const className = "movie-menu-close";
    const title = "Close";
    const component = shallow(
      <MovieMenuOpen action={action}
                     editMovie={editMovie}
                     deleteMovie={deleteMovie} />
    )
    const button = component.find("#movie-menu-close-button");
    button.simulate('click');
    expect(button.hasClass(className)).toEqual(true);
    expect(button.text()).toEqual(title);
  });

});
