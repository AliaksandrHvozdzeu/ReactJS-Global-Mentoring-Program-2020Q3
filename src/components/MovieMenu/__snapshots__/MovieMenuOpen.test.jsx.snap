// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when MovieMenuOpen then snapshot created 1`] = `
<div
  className="movie-menu movie-menu-opened"
>
  <button
    className="movie-menu-action"
    onClick={[Function]}
    type="button"
  >
    Edit
  </button>
  <button
    className="movie-menu-action"
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
  <button
    className="movie-menu-close"
    id="movie-menu-close-button"
    onClick={[Function]}
    type="button"
  >
    Close
  </button>
</div>
`;
